### ISSUE TITLE
Option to remove Grammarly and Spotify 

### ISSUE THREAD

**Issue by projektilski** [Raphire/Win11Debloat#41]
Hi, can you implement an option to remove Grammarly and Spotify?

**Comment by Raphire** [comment]
Heya,

Could you confirm that they're actually installed on the system? My guess is that they're actually just pinned to the start menu, similar to the situation from your other issue #40.

**Comment by Raphire** [comment]
It's interesting that the behaviour is so different between the 3 apps. 

Could you confirm if they're actually recognized as installed and removable by running the following commands in powershell:
Spotify: `Get-AppxPackage *spotify*`
![image](https://github.com/Raphire/Win11Debloat/assets/9938813/92e1825f-e329-456d-a7aa-b0f5a6a19db1)

LinkedIn: `Get-AppxPackage *linkedin*`
![image](https://github.com/Raphire/Win11Debloat/assets/9938813/70f6dea3-386e-45ea-a65c-2974a8455f35)

Grammarly: `winget list grammarly`
![image](https://github.com/Raphire/Win11Debloat/assets/9938813/bd65f559-b3ab-4d17-bc92-af43ec1e5dbf)

**Comment by Raphire** [comment]
I have done some additional testing by doing a few fresh installs. In those installs Grammarly, LinkedIn, Spotify, and Luminar Neo do show up in the start menu, but none are installed. I also couldn't find any files related to Spotify, or any of the other 3 apps.

**Comment by Raphire** [comment]
> Hi, I did some testing also. If I understand correctly, your script removes apps for all users. When I create a new user on the same machine, all those apps are listed in Start Menu, but are not actually installed. Right click on them offers uninstall option, but what it actually does is to remove the pinned item.
So the real issue is how to remove those unwanted pinned items for new users.

Exactly right, the rightclick menu does act as if they are installed (as they show an uninstall option), but they're not. 

The annoying part is that these app links are, at least for a script like this, much harder to remove than the actual apps if they were installed. I am continuing to look into ways to have the script be able to unpin individual apps, but right now the only way I have found that works is replacing the whole start menu layout all together.

> p.s. I found a new app Journal which is not pinned in Start Menu but it is listed under All apps. It can be uninstalled via Powershell using Microsoft.MicrosoftJournal manifes name.

I'll be adding support for this app to the script, thanks for bringing that to my attention 😀

**Comment by Raphire** [comment]
I've renamed the issue (and definitely didn't make a spelling mistake) to better reflect the feature requested.

**Comment by miladhatami1393** [comment]
Hi, I did some testing also. If I understand correctly, your script removes apps for all users. When I create a new user on the same machine, all those apps are listed in Start Menu, but are not actually installed. Right click on them offers uninstall option, but what it actually does is to remove the pinned item.
So the real issue is how to remove those unwanted pinned items for new users.

p.s. I found a new app Journal which is not pinned in Start Menu but it is listed under All apps. It can be uninstalled via Powershell using Microsoft.MicrosoftJournal manifes

**Comment by Raphire** [comment]
> Hi, I did some testing also. If I understand correctly, your script removes apps for all users. When I create a new user on the same machine, all those apps are listed in Start Menu, but are not actually installed. Right click on them offers uninstall option, but what it actually does is to remove the pinned item. So the real issue is how to remove those unwanted pinned items for new users.

That's correct, the items are not installed so they cannot be removed. They only exist as a pinned item in the start menu binary. This binary (start2.bin) is stored per user at: `C:\Users\{USERNAME}\AppData\Local\Packages\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy\LocalState`

As the file is a binary it cannot be edited easily. That's why Win11Debloat can only clear the entire start menu by replacing this file with another.

> p.s. I found a new app Journal which is not pinned in Start Menu but it is listed under All apps. It can be uninstalled via Powershell using Microsoft.MicrosoftJournal manifes

Win11Debloat should be able to uninstall this already. It's actually marked to be uninstalled as part of the default setting.

**Comment by sean-sauve** [comment]
@Raphire Instead of using start2.bin from another user, you could use Export-StartLayout and Import-StartLayout.  That would make it easier for people to review and change over a bin file.

https://learn.microsoft.com/en-us/powershell/module/startlayout/import-startlayout

https://learn.microsoft.com/en-us/windows/configuration/start/layout?tabs=intune-10%2Cintune-11&pivots=windows-11

You could also have a "remove" function that exports the current start menu, opens the json and removes the lines for Spotify and saves it, and then imports the modified layout.  Cost/benefit might be low on building that, but by working with structured data that Posh can understand it opens up more possibilities.  These json files are also how Intune works to apply start menu configs, so you'd be in good company.

**Comment by Raphire** [comment]
> @Raphire Instead of using start2.bin from another user, you could use Export-StartLayout and Import-StartLayout. That would make it easier for people to review and change over a bin file.
> 
> https://learn.microsoft.com/en-us/powershell/module/startlayout/import-startlayout
> 
> https://learn.microsoft.com/en-us/windows/configuration/start/layout?tabs=intune-10%2Cintune-11&pivots=windows-11
> 
> You could also have a "remove" function that exports the current start menu, opens the json and removes the lines for Spotify and saves it, and then imports the modified layout. Cost/benefit might be low on building that, but by working with structured data that Posh can understand it opens up more possibilities. These json files are also how Intune works to apply start menu configs, so you'd be in good company.

Unfortunately I don't think the `Import-StartLayout` cmdlet works for Windows 11. 

I'm sure there are other ways to apply the layout you can export with the `Export-StartLayout` cmdlet, but the way the supported management tools like InTune do this is through policies. According to[ the page you linked](https://learn.microsoft.com/en-us/windows/configuration/start/layout?tabs=intune-10%2Cintune-11&pivots=windows-11#user-experience) this would mean the start menu layout cannot be changed anymore by the user, quote: 
>Users can change the order of the pinned elements, pin, or unpin items. When a user signs in again, the Start layout specified in the policy setting is reapplied, without retaining any user changes.

If there is a different way to apply a Start layout I'd be happy to implement it into the script.

**Comment by sean-sauve** [comment]
Good point, I didn't know this didn't work for Windows 11 sorry.

**Comment by Raphire** [comment]
> Good point, I didn't know this didn't work for Windows 11 sorry.

No worries! I appreciate you taking the time to look into this issue, it's just a shame Microsoft locked simple stuff like this down.
